src/nvim/fileio.c

```diff
@@ -4333,108 +4333,95 @@ void shorten_fnames(int force)
+/// Returns NULL when out of memory.
+/// @param fname When NULL use the current directory name
+/// @param ext The extension to add to the filename, 4 chars max if prefixed 
+///            with a dot, 3 otherwise.
+/// @param prepend_dot If true, prefix the ext with a dot
+char *modname (char *fname, char *ext, bool prepend_dot) 
 {
-  char_u      *retval;
-  char_u      *s;
-  char_u      *e;
-  char_u      *ptr;
-  int fnamelen, extlen;
-
-  extlen = (int)STRLEN(ext);
+  char *retval;
+  size_t fnamelen;
+  size_t extlen = strlen(ext);
 
-  /*
-   * If there is no file name we must get the name of the current directory
-   * (we need the full path in case :cd is used).
-   */
+  // If there is no file name we must get the name of the current directory
+  // (we need the full path in case :cd is used).
   if (fname == NULL || *fname == NUL) {
     retval = xmalloc(MAXPATHL + extlen + 3);
-    if (os_dirname(retval, MAXPATHL) == FAIL ||
-        (fnamelen = (int)STRLEN(retval)) == 0) {
+    if (os_dirname((char_u *)retval, MAXPATHL) == FAIL ||
+        (fnamelen = strlen(retval)) == 0) {
       xfree(retval);
       return NULL;
     }
-    if (!after_pathsep((char *)retval, (char *)retval + fnamelen)) {
+    if (!after_pathsep(retval, retval + fnamelen)) {
       retval[fnamelen++] = PATHSEP;
       retval[fnamelen] = NUL;
     }
-    prepend_dot = FALSE;            /* nothing to prepend a dot to */
+    prepend_dot = FALSE;  // nothing to prepend a dot to
   } else {
-    fnamelen = (int)STRLEN(fname);
+    fnamelen = strlen(fname);
     retval = xmalloc(fnamelen + extlen + 3);
-    STRCPY(retval, fname);
+    strcpy(retval, fname);
   }
 
-  /*
-   * search backwards until we hit a '/', '\' or ':'.
-   * Then truncate what is after the '/', '\' or ':' to BASENAMELEN characters.
-   */
+  // Search backwards until we hit a '/', '\' or ':'.
+  // Then truncate what is after the '/', '\' or ':' to BASENAMELEN characters.
+  char *ptr = NULL;
   for (ptr = retval + fnamelen; ptr > retval; mb_ptr_back(retval, ptr)) {
     if (vim_ispathsep(*ptr)) {
-      ++ptr;
+      ptr++;
       break;
     }
   }
 
-  /* the file name has at most BASENAMELEN characters. */
-  if (STRLEN(ptr) > BASENAMELEN)
+  // the file name has at most BASENAMELEN characters.
+  if (strlen(ptr) > BASENAMELEN) {
     ptr[BASENAMELEN] = '\0';
+  }
 
-  s = ptr + STRLEN(ptr);
+  char *s;
+  s = ptr + strlen(ptr);
 
 #if defined(WIN3264)
-  /*
-   * If there is no file name, and the extension starts with '.', put a
-   * '_' before the dot, because just ".ext" may be invalid if it's on a
-   * FAT partition, and on HPFS it doesn't matter.
-   */
-  else if ((fname == NULL || *fname == NUL) && *ext == '.')
+  // If there is no file name, and the extension starts with '.', put a
+  // '_' before the dot, because just ".ext" may be invalid if it's on a
+  // FAT partition, and on HPFS it doesn't matter.
+  else if ((fname == NULL || *fname == NUL) && *ext == '.') {
```

Comment:
Future note: This pattern `(pointer == NULL || *pointer == NUL)` is repeated over and over again. Not here, but at some point, it would be a nice refactoring replacing this by an `empty(pointer)` inline function (better than macro).